packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -467,6 +441,37 @@ class ThemeData {
     );
   }
 
+  // See <https://www.w3.org/TR/WCAG20/#relativeluminancedef>
+  static double _linearizeColorComponent(double component) {
+    if (component <= 0.03928)
+      return component / 12.92;
+    return math.pow((component + 0.055) / 1.055, 2.4);
+  }
+
+  /// Determines whether the given [Color] is [Brightness.light] or
+  /// [Brightness.dark].
+  ///
+  /// This compares the luminosity of the given color to a threshold value that
+  /// matches the material design specification.
+  static Brightness estimateBrightnessForColor(Color color) {
+    // See <https://www.w3.org/TR/WCAG20/#relativeluminancedef>
+    final double R = _linearizeColorComponent(color.red / 0xFF);
```

Comment:
I'm surprised that it's necessary to mask of the lower 8 bits. I thought Color.foo already masked off 8 bits